### ISSUE TITLE
Add prediction sample code to text classifier

### ISSUE THREAD

**Issue by DGaffney** [fastai/fastai#873]
**Describe the bug**
The tutorial explaining a text classifier based on IMDB data stops just short of actually being used for anything beyond the fitting stage: https://github.com/fastai/fastai/blob/master/examples/text.ipynb. Is there any way we can get demo code for predictions as well?

**To Reproduce**
Running this code leaves you with, presumably, a fit model, but no way to interact with it afterwards. Based on scanning around, I have hobbled together a (probably wrong) set of commands to run that would produce results from the model - is this at all close to how one would be expected to use the models, and if not, is it possible for the fastai team to make prediction as simple as model generation?

```python
from fastai.text import *
from torch.autograd.variable import Variable
def softmax(x):
    if x.ndim == 1:
        x = x.reshape((1, -1))
    max_x = np.max(x, axis=1).reshape((-1, 1))
    exp_x = np.exp(x - max_x)
    return exp_x / np.sum(exp_x, axis=1).reshape((-1, 1))

model = learn.classifier(data_clas).model
model.reset()
model.eval()
df = pd.read_csv(IMDB_PATH/'valid.csv', header=None)
stoi = data_lm.train_ds.vocab.stoi
results = []
for text in df[1]:
    texts = [text]
    tok = Tokenizer().process_all(texts)
    encoded = [stoi[p] for p in tok[0]]
    ary = np.reshape(np.array(encoded),(-1,1))
    tensor = torch.from_numpy(ary)
    variable = Variable(tensor)
    predictions = model(variable)
    numpy_preds = predictions[0].data.numpy()
    score = softmax(numpy_preds[0])[0].tolist()
    print(score)
    results.append(score)
```

**Comment by cfotache** [comment]
Maybe the same for the vision classifier, in the MNIST and Cats or Dogs examples?

**Comment by sgugger** [comment]
You need to add a few lines of code, yes, though you can refactor quite a bit your code:
`learn.get_preds()` would give you the outputs of the model concatenated. Then `F.softmax` can be used to compute the softmax before putting everything in results.
Please switch to the [forum](http://forums.fast.ai) if you have more questions.
